services-directions-models/src/main/java/com/mapbox/api/directions/v5/utils/FormatUtils.java

```diff
@@ -125,12 +125,26 @@ public static String formatBearings(@Nullable List<List<Double>> bearings) {
 
     List<String> bearingsToJoin = new ArrayList<>();
     for (List<Double> bearing : bearings) {
-      if (bearing == null || bearing.size() == 0) {
+      if (bearing == null) {
         bearingsToJoin.add(null);
       } else {
-        bearingsToJoin.add(String.format(Locale.US, "%s,%s",
-            formatCoordinate(bearing.get(0)),
-            formatCoordinate(bearing.get(1))));
+        if (bearing.size() != 2) {
+          throw new RuntimeException("Bearing size should be 2.");
+        }
+
+        Double angle = bearing.get(0);
+        Double tolerance = bearing.get(1);
+        if (angle == null || tolerance == null) {
+          bearingsToJoin.add(null);
```

Comment:
I feel like we're safe to throw an exception here. If we chose to provide angle, we have to provide a tolerance as well. And vice versa.

Reply:
Ah now I see that the `addBearing` method accepts nulls for angle and tolerance. Not sure if that's correct, but I guess we can leave this as is.

Reply:
ok, resolve for now